src/main/java/com/google/step/servlets/UserServlet.java

```diff
@@ -58,4 +62,36 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json");
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    if (!userService.isUserLoggedIn()) {
+      response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+      return;
+    }
+
+    String idString;
+    try {
+      idString = request.getPathInfo().substring(1);
+    } catch (IndexOutOfBoundsException e) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    }
+    String userEmail = userService.getCurrentUser().getEmail();
+    String username = (String) request.getParameter("username");
+    String bio = (String) request.getParameter("bio");
+
+    User user = userManager.readOrCreateUserByEmail(userEmail);
+
+    if (!idString.equals(String.valueOf(user.id))) {
+      // Inconsistent request with login status
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    }
```

Comment:
Over here, if I were to pass an invalid user ID in my POST request, you would still end up creating the user through `readOrCreateUserByEmail()`? Maybe there should be a read-only method that you invoke instead?